### PR TITLE
:green_heart: add eslint in ci

### LINKED ISSUE #329: Disable merge if eslint fail
As developer, I want push my code with no errors in order to keep the project clean 

add eslint in config CI 

Rules : 
- max warning at 0 
- unused variables

### PR DESCRIPTION
**Pull request by MailyLehoux**
resolve #329

### CODE REVIEW

File: src/components/CustomSelect.tsx [hunk 1]
```diff
@@ -4,12 +4,12 @@ import styles from "./CustomSelect.module.css";
 
 type CustomSelectProps = {
   choices: any[];
-  keyFn: (x: any) => string;
-  labelFn: (x: any) => string;
+  keyFn: (x: any) => string; // eslint-disable-line no-unused-vars
```
Comment (PierreBrethes) [inline src/components/CustomSelect.tsx:7]: Faudrait voir si il n'y a pas une vrai correction à la place de l'exception lint

**Review by benjaminbrgs — CHANGES_REQUESTED** [review]
The `next build` command already includes the eslint process.
Is it necessary to add this step a second time ?

**Comment by PierreBrethes** [comment]
> The `next build` command already includes the eslint process. Is it necessary to add this step a second time ?
> 
> Otherwise, this PR is welcome to fix some eslint errors

This make sure that when you push code, it's clean. I think this is good for the quality code
And it takes few seconds

**Comment by MailyLehoux** [comment]
> The `next build` command already includes the eslint process. Is it necessary to add this step a second time ?
> 
> Otherwise, this PR is welcome to fix some eslint errors

We had the basic eslint, I've added also some rules. And the eslint tests weren't on ci jobs. I don't think that eslint was enclenched at each push

**Comment by benjaminbrgs** [comment]
> > The `next build` command already includes the eslint process. Is it necessary to add this step a second time ?
> > Otherwise, this PR is welcome to fix some eslint errors
> 
> We had the basic eslint, I've added also some rules. And the eslint tests weren't on ci jobs. I don't think that eslint was enclenched at each push

> This make sure that when you push code, it's clean. I think this is good for the quality code And it takes few seconds


I agree with adding some rules 👍🏻 
I mean, is it not a duplicate to add a eslint step in job as it is already running in another step (see https://app.circleci.com/pipelines/github/Zenika/skillZ/956/workflows/815b382d-6e7a-43b0-a7c3-a27a16637fc4/jobs/4817?invite=true#step-107-213) ?

**Comment by MailyLehoux** [comment]
Indeed ! I didn't noticed that runs in next build. it looks like it don't take in account the new rules with this way. Maybe should we delete the eslint step from build ? And keep the eslint job.  For me it's clearer

**Comment by benjaminbrgs** [comment]
> Indeed ! I didn't noticed that runs in next build. it looks like it don't take in account the new rules with this way. Maybe should we delete the eslint step from build ? And keep the eslint job. For me it's clearer

Agree with this proposal, it will be clearer

File: src/components/BotNotifications.tsx [hunk 2]
```diff
@@ -7,9 +7,13 @@ import { useQuery } from "@apollo/client";
 import { GetBotNotificationsQuery } from "../generated/graphql";
 import Loading from "./Loading";
 
-export const BotNotifications = ({ t }: { t: (path: string) => string }) => {
+export const BotNotifications = ({
+  t,
+}: {
+  t: (path: string) => string /* eslint-disable-line no-unused-vars*/;
```
Comment (benjaminbrgs) [inline src/components/BotNotifications.tsx:13]:
```suggestion
  t: (path: string) => string;
```

**Review by benjaminbrgs — APPROVED** [review]
Nice job 💪🏼

File: src/components/BotNotifications.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MailyLehoux) [reply src/components/BotNotifications.tsx]: Oops I missed this, nice catch thanks :+1:

**Pull request merged**